### DISCUSSION TITLE
"Not found" error

### DISCUSSION THREAD

**Discussion by Roomiiaan** [YOURLS/YOURLS#3028] [Q&A]
Hey guys, I'm sure this is not the first time this question has been asked.

Unfortunately, when I want to call up a URL, I have the problem that I always come to the "Not found" page (not being redirected)

![grafik](https://user-images.githubusercontent.com/31469387/130355966-cc344528-ecf7-49bd-b1f5-65030bb15767.png)

My `.htaccess` file
```
# BEGIN YOURLS
<IfModule mod_rewrite.c>
RewriteEngine On
RewriteBase /go/
RewriteCond %{HTTPS} !=on
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^.*$ /go/yourls-loader.php [L]
</IfModule>
# END YOURLS
```

Maybe anyone can help me. Thanks!

**Comment by logusivam** [comment]
Your `RewriteBase /go/` is correct for a subdirectory install — that part is fine. The problem is a misplaced `RewriteCond` that is blocking redirects for HTTPS visitors.

---

## The bug — `RewriteCond %{HTTPS} !=on` is in the wrong place and doing the wrong thing

```apache
RewriteCond %{HTTPS} !=on
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^.*$ /go/yourls-loader.php [L]
```

All three `RewriteCond` lines apply to the **same** `RewriteRule` below them — Apache requires **all conditions** to be true simultaneously for the rule to fire. By including `RewriteCond %{HTTPS} !=on` here, you are telling Apache:

> "Only send requests to `yourls-loader.php` if the connection is **NOT** HTTPS."

So if your site runs on HTTPS (which is almost certainly the case), **this condition is always false**, the rule never fires, and every short URL hits the "Not found" page instead of being processed by YOURLS. This explains your symptom exactly.

That condition belongs in a separate rule block for HTTP→HTTPS redirection, not as a gate on YOURLS's core redirect rule.

---

## The fix

**If you want to force HTTPS (recommended):**

```apache
# BEGIN YOURLS
<IfModule mod_rewrite.c>
RewriteEngine On
RewriteBase /go/

# Rule 1: Force HTTP → HTTPS
RewriteCond %{HTTPS} !=on
RewriteRule ^ https://%{HTTP_HOST}%{REQUEST_URI} [L,R=301]

# Rule 2: Route all non-file, non-directory requests through YOURLS
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^.*$ /go/yourls-loader.php [L]
</IfModule>
# END YOURLS
```

**If you don't need HTTP→HTTPS forcing and just want YOURLS to work:**

```apache
# BEGIN YOURLS
<IfModule mod_rewrite.c>
RewriteEngine On
RewriteBase /go/
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^.*$ /go/yourls-loader.php [L]
</IfModule>
# END YOURLS
```

Simply remove the `RewriteCond %{HTTPS} !=on` line entirely — it serves no purpose in this position and actively breaks HTTPS visitors.

---

## Why this happens

`RewriteCond` in Apache is not a standalone directive — it always attaches to the very next `RewriteRule`. When you stack multiple `RewriteCond` lines before one `RewriteRule`, Apache treats them as a logical AND: every single condition must be true for the rule to execute. Placing an HTTPS-check condition in the same block as your YOURLS routing rule means YOURLS only routes non-HTTPS traffic, silently doing nothing for everyone else.

The correct pattern is always:

```apache
# Each RewriteCond block + RewriteRule is a self-contained unit
RewriteCond %{HTTPS} !=on          ← applies only to this rule ↓
RewriteRule ^ https://...  [R=301,L]

RewriteCond %{REQUEST_FILENAME} !-f  ← applies only to this rule ↓
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^.*$ /go/yourls-loader.php [L]
```

---

## After saving the fix

1. Save the corrected `.htaccess` file.
2. Clear any browser cache for your short URL domain.
3. Test by visiting one of your short URLs — it should now redirect correctly.
4. If you added the HTTP→HTTPS rule, test an `http://` URL to confirm it upgrades to `https://` before redirecting.

Hope this helps — if it answers your question, would you mind clicking **"Mark as answer"** so others searching for the same thing can find it quickly?

**Comment by ozh** [comment]
Hey, @logusivam aka Mr ChatGPT is back.

Dear @logusivam,

We appreciate your effort to solve as many discussions as possible. BUT : 
- I'm not sure there's a point in replying to 5 year old discussions ?
- Your answers are WAY TOO LONG. Your LLM sounding (because they are 100% LLM, right?) replies are 10 times longer than necessary. The correct answer here is probably "you put that line in the wrong place", and not a 30 lines essay.
- Your answers are OFF TOPIC. Explaining how `.htaccess` works, or explaining how to enforce https, are not parts of the initial question. You're answering questions not raised. You're mixing up "helping 1 problem" with "writing a guide about 100 stuff".
- Your way too long answers are emailed to hundreds of people watching this repo.

Please keep your answers short and to the point. And please understand there won't another "please keep your answers short" notice.
